python/cuml/linear_model/linear_regression_mg.pyx

```diff
@@ -23,610 +22,371 @@ import ctypes
-        algorithm : Type: string. 'eig' (default) and 'svd' are supported
-        algorithms.
-        fit_intercept: boolean. For more information, see `scikitlearn's OLS
-        <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
-        normalize: boolean. For more information, see `scikitlearn's OLS
-        <https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html>`_.
+from libc.stdlib cimport malloc, free
 
-        """
-        self.coef_ = None
-        self.intercept_ = None
-        self.fit_intercept = fit_intercept
-        self.normalize = normalize
-        if algorithm in ['svd', 'eig']:
-            self.algo = self._get_algorithm_int(algorithm)
-        else:
-            msg = "algorithm {!r} is not supported"
-            raise TypeError(msg.format(algorithm))
-
-        self.intercept_value = 0.0
-
-    def _get_algorithm_int(self, algorithm):
-        return {
-            'svd': 0,
-            'eig': 1
-        }[algorithm]
-
-    def _get_ctype_ptr(self, obj):
-        # The manner to access the pointers in the gdf's might change, so
-        # encapsulating access in the following 3 methods. They might also be
-        # part of future gdf versions.
-        return obj.device_ctypes_pointer.value
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t, uint32_t, uint64_t
+from cython.operator cimport dereference as deref
+
+from cuml.common.base import Base
+from cuml.common.handle cimport cumlHandle
+from cuml.decomposition.utils cimport *
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros
+
+from cuml.linear_model import LinearRegression
+
+cdef extern from "cumlprims/opg/matrix/data.hpp" \
+                 namespace "MLCommon::Matrix":
+
+    cdef cppclass floatData_t:
+        floatData_t(float *ptr, size_t totalSize)
+        float *ptr
+        size_t totalSize
+
+    cdef cppclass doubleData_t:
+        doubleData_t(double *ptr, size_t totalSize)
+        double *ptr
+        size_t totalSize
+
+cdef extern from "cumlprims/opg/matrix/part_descriptor.hpp" \
+                 namespace "MLCommon::Matrix":
+
+    cdef cppclass RankSizePair:
+        int rank
+        size_t size
+
+cdef extern from "cumlprims/opg/ols.hpp" namespace "ML::OLS::opg":
+
+    cdef void fit(cumlHandle& handle,
+		  RankSizePair **rank_sizes,
+		  size_t n_parts,
+		  floatData_t **input,
+		  size_t n_rows,
+		  size_t n_cols,
+		  floatData_t **labels,
+		  float *coef,
+		  float *intercept,
+		  bool fit_intercept,
+		  bool normalize,
+                  int algo,
+		  bool verbose) except +
+
+    cdef void fit(cumlHandle& handle,
+		  RankSizePair **rank_sizes,
+		  size_t n_parts,
+		  doubleData_t **input,
+		  size_t n_rows,
+		  size_t n_cols,
+		  doubleData_t **labels,
+		  double *coef,
+		  double *intercept,
+		  bool fit_intercept,
+		  bool normalize,
+                  int algo,
```

Comment:
There are two PRs, one for OLS and one for Ridge. linear_regression_mg.pyx file belongs to OLS PR and all these type of things were already fixed. Otherwise, flake8 would complain.